src/include/task/task_manager.h

```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+#include "common/dedicated_thread_owner.h"
+#include "task/task.h"
+#include "task/task_runner.h"
+
+#include <condition_variable>
+#include <mutex>
+#include <queue>
+#include <vector>
+
+namespace noisepage::util {
+class QueryExecUtil;
+}
+
+namespace noisepage::task {
+
+class TaskRunner;
+
+class TaskManager : public common::DedicatedThreadOwner {
+ public:
+  TaskManager(common::ManagedPointer<common::DedicatedThreadRegistry> thread_registry,
+              std::unique_ptr<util::QueryExecUtil> util, uint32_t num_workers);
+
+  ~TaskManager();
+
+  void Flush();
```

Comment:
Seems that `Flush` only waits for things to finish, instead of trying to actively flush things? If so, it would be good to clarify.